pod-work.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  creationTimestamp: null
+  labels:
+    run: abcdefg
+    feeling: ThisWorldIsUnfirendly
+  name: abcdefg
+spec:
+  containers:
+  - image: mysql:8.0.27
+    name: mysql
+    env:
+      - name: MYSQL_ROOT_PASSWORD
+        value: 'root-password'
+      - name: MYSQL_DATABASE
+        value: 'wp-database'
+      - name: MYSQL_USER
+        value: 'wp-user'
+      - name: MYSQL_PASSWORD
+        value: 'wp-password'
+    livenessProbe:
+      tcpSocket:
+        port: 3306
+      initialDelaySeconds: 60
+      periodSeconds: 15
+      timeoutSeconds: 15
+      successThreshold: 1
+      failureThreshold: 3
+    resources:
+      requests:
+        memory: "64Mi"
+        cpu: "250m"
+      limits:
+        memory: "512Mi"
+        cpu: "500m"
+  - image: wordpress:5.8.2-apache
+    name: wordpress
+    env:
+      - name: WORDPRESS_DB_HOST
+        value: '127.0.0.1:3306'
+      - name: WORDPRESS_DB_USER
+        value: 'wp-user'     
+      - name: WORDPRESS_DB_PASSWORD
+        value: 'wp-password'
+      - name: WORDPRESS_DB_NAME
+        value: 'wp-database'
+    livenessProbe:
+      httpGet:
+        path: /
+        port: 80
+        httpHeaders:
+        - name: Custom-Header
+          value: Awesome
+      initialDelaySeconds: 300
+      periodSeconds: 10
+      timeoutSeconds: 15
+      successThreshold: 1
+      failureThreshold: 3
+    readinessProbe:
+      tcpSocket:
+        port: 3306
+      initialDelaySeconds: 60
+      periodSeconds: 15
+      timeoutSeconds: 15
+      successThreshold: 1
+      failureThreshold: 3
+    resources:
+      requests:
+        memory: "64Mi"
+        cpu: "250m"
+      limits:
+        memory: "512Mi"
+        cpu: "500m"
+  dnsPolicy: ClusterFirst
+  restartPolicy: OnFailure
+status: {}
+# 所有探針加上 host: 127.0.0.1都會掛掉
```

Comment:
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#tcp-probes

```
For a TCP probe, the kubelet makes the probe connection at the node, not in the pod, which means that you can not use a service name in the host parameter since the kubelet is unable to resolve it.
```

tcpSocket 與 httpGet 是由該節點的 kubelet 在容器外發起 request，因此填 127.0.0.1 會變成 kubelet 檢測自己那個節點。